doc/aaa/RBAC_Phase2.md

```diff
@@ -0,0 +1,421 @@
+
+This service runs a process that uses the POSIX [inotify APIs](http://man7.org/linux/man-pages/man7/inotify.7.html) to register for file system events like changes to `/etc/passwd` and/or `/etc/shadow`.
+Another approach would be to have a process started by `systemd` on changes to `/etc/passwd` or `/etc/group`, and that process would simply reconcile the Redis DB with what is found in those files. `systemd` allows starting processes based on file create/delete/modify.
+
+A user can be created either via CLI or REST. It is the responsibility of this service to ensure that when the user information is added to the User DB, the appropriate user and groups are also created in the `/etc/passwd` and `/etc/groups` files.
+This way, the User DB information and the Linux groups information is always in sync.
+
+## 3.5 SAI
+N/A
+
+## 3.6 User Interface
+### 3.6.1 Data Models
+TBD from developer
+(TODO/DELL)
+
+### 3.6.2 CLI
+#### 3.6.2.1 Configuration Commands for User Management
+Users may be managed via Linux tools like `useradd`, `usermod`, `passwd`, etc. They may also be managed via configuration.
+
+##### username
+`username <name> password <password-string> role <role-string>` -- Configures a user on the system with a given name, password, and role.
+* **name** is a text string of 1-32 alphanumeric characters
+* **password-string** is a text string of 1-32 alphanumeric characters
+* **role-string** is a text string consisting of a role name. In the initial release, the user is recommended to use "admin" and "operator" roles, as other roles will not be supported. A text string is desired instead of keywords so that in the future, more roles may be implemented and expanded.
+* Configuring another a user with the same **name** should result in modification of the existing user.
+
+`no username <name>` -- Deletes a user from the system.
+* **name** is a text string of 1-32 alphanumeric characters
+
+#### 3.6.2.2 Show Commands
+N/A
+
+#### 3.6.2.3 Debug Commands
+N/A
+
+#### 3.6.2.4 IS-CLI Compliance
+N/A
+
+### 3.6.3 REST API Support
+Ability to configure local users via REST API.
+
+# 4 Flow Diagrams
+N/A
+
+# 5 Error Handling
+## 5.1 REST Server
+The REST server should return standard HTTP errors when authentication fails or if the user tries to access a forbidden resource or perform an unauthorized activity.
+
+## 5.2 gNMI server
+The gNMI server should return standard gRPC errors when authentication fails.
+
+## 5.3 CLI
+Authentication errors will be handled by SSH. However, the CLI must gracefully handle authorization failures from the REST server (the authorization failure would originate from Translib of course). While the CLI will render all of the available commands to a user, the user will actually only be able to execute a subset of them. This limitation is a result of the design decision to centralize RBAC in Translib. Nevertheless, the CLI must inform the user when they attempt to execute an unauthorized command.
```

Comment:
If possible, we should also write a log entry when a user tries to execute an unauthorized command.